### PR TITLE
Push the redemption rate to Oracle contract V2

### PR DESCRIPTION
**Pull request by rustcandy**
## Overview

Fix for pushing proper calculated redemption rate to Oracle contract v2

closes: #XXXX

## What changes have been made in this PR?

- [ ]

## Checklist

---

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation

### CODE REVIEW

File: contracts/staking/src/helpers.rs [hunk 1]
```diff
@@ -151,14 +151,16 @@ where
     Ok(result)
 }
 
-pub fn get_redemption_rate(deps: &Deps) -> Decimal {
+pub fn get_rate(deps: &Deps) -> (Decimal, Decimal) {
```
Comment (hallazzang) [inline contracts/staking/src/helpers.rs:154]: I suggest the name `get_rates` instead

File: contracts/staking/src/query.rs [hunk 2]
```diff
@@ -55,11 +55,11 @@ pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
 
 pub fn query_state(deps: Deps) -> StdResult<StateResponse> {
     let state = STATE.load(deps.storage)?;
-
+    let (redemption_rate, _purchase_rate) = get_rate(&deps);
     let res = StateResponse {
         total_native_token: state.total_native_token,
         total_liquid_stake_token: state.total_liquid_stake_token,
-        rate: get_redemption_rate(&deps),
+        rate: redemption_rate,
```
Comment (hallazzang) [inline contracts/staking/src/query.rs:62]: I think this should be `purchase_rate` in order to not break backward compatibility.

File: contracts/staking/src/query.rs [hunk 3]
```diff
@@ -55,11 +55,11 @@ pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
 
 pub fn query_state(deps: Deps) -> StdResult<StateResponse> {
     let state = STATE.load(deps.storage)?;
-
+    let (redemption_rate, _purchase_rate) = get_rate(&deps);
```
Comment (hallazzang) [inline contracts/staking/src/query.rs:58]:
```suggestion
    let (redemption_rate, _) = get_rate(&deps);
```

Note: we should `purchase_rate` according to the comment below

**Review by hallazzang — CHANGES_REQUESTED** [review]

File: contracts/staking/src/query.rs (hunk 2 above)
Comment (rustcandy) [reply contracts/staking/src/query.rs:62]: Yes nice catch

**Review by hallazzang — APPROVED** [review]
LGTM

**Review by RiccardoM — APPROVED** [review]
Looks good to me

**Pull request merged**